be/src/exec/olap_scan_node.cpp

```diff
@@ -1250,12 +1250,25 @@ void OlapScanNode::transfer_thread(RuntimeState* state) {
 
     state->resource_pool()->release_thread_token(true);
     VLOG(1) << "TransferThread finish.";
-    std::unique_lock<std::mutex> l(_row_batches_lock);
-    _transfer_done = true;
+    {
+        std::unique_lock<std::mutex> l(_row_batches_lock);
+        _transfer_done = true;
+    }
     _row_batch_added_cv.notify_all();
+
+    std::unique_lock<std::mutex> l(_scan_batches_lock);
+    while (_running_thread != 0) {
```

Comment:
Is it better to use C++11 promise and future to instead of checking _running_thread == 0? every scanner_thread just need to set its promise value when finish task with lock free, and close OlapScanNode close just need to wait all scanner_thread return future result ?